process/process.go

```diff
@@ -65,8 +81,18 @@ func trimMappingPath(path string) string {
 func parseMappings(mapsFile io.Reader) ([]Mapping, error) {
 	mappings := make([]Mapping, 0)
 	scanner := bufio.NewScanner(mapsFile)
-	buf := make([]byte, 512)
-	scanner.Buffer(buf, 8192)
+	scanBuf := bufPool.Get().(*[]byte)
+	if scanBuf == nil {
+		return mappings, errors.New("failed to get memory from sync pool")
+	}
+	defer func() {
+		// Reset memory and return it for reuse.
+		for j := 0; j < len(*scanBuf); j++ {
```

Comment:
```suggestion
		for j := len(*scanBuf) {
```

Reply:
While this new syntax should also work, I don't see a benefit here. While the old syntax uses 12 more characters, it is easier to determine boundaries of the loop(e.g. is `len(*scanBuf)` included or excluded in `j := len(*scanBuf)`?) and the size of the incremental steps.

Reply:
The Go range loop is well defined. It goes from `0..N-1` in the construct `for range N`.

Reply:
See also https://cloud-native.slack.com/archives/C07CCCMRXBK/p1739546386809279

Reply:
Oh, just saw the suggestion was missing the `range`. Fixed now.

Reply:
I'm not entirely convinced on the suggested change to `for j := range *scanBuf `. It does not improve performance and has the chance to introduce off-by-one issues, if not consulting (or know by heart) the Go language specification.

Reply:
Not a blocker :)

Reply:
We could use `copy` which could be better performance-wise, though probably overkill for this specific use-case:

https://gist.github.com/taylorza/df2f89d5f9ab3ffd06865062a4cf015d